### PR TITLE
feat:adds the grid component

### PR DESCRIPTION
**Pull request by FarhanMobashir**
grid-4
grid-res
-adds a basic 4 column grid system
-adds a resposive 4 column grid

### CODE REVIEW

**Review by GuptaEesh — COMMENTED** [review]
Clean Code. Minimal usage and fluid layout. Looks great✅

File: src/components/grid.css [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+.grid-component-responsive {
+  display: grid;
+  grid-template-columns: repeat(4, 1fr);
+  grid-template-rows: auto;
+  gap: 1rem;
+}
+
+@media (max-width: 35rem) {
```
Comment (Hamdrive) [inline src/components/grid.css]: You can use `px` here, this would make it easier to know the breaking point in media query

**Review by Hamdrive — COMMENTED** [review]
Good work done, implementation looks nice. Have suggested for media query

File: src/components/grid.css (hunk 1 above)
Comment (FarhanMobashir) [reply src/components/grid.css]: Sure . Will change those

**Pull request merged**